src/viam/ur/module/ur_arm.cpp

```diff
@@ -620,36 +620,36 @@ void URArm::move(std::vector<Eigen::VectorXd> waypoints, std::chrono::millisecon
             throw std::runtime_error(buffer.str());
         }
 
-        const float duration = static_cast<float>(trajectory.getDuration());
+        const double duration = trajectory.getDuration();
         if (std::isinf(duration)) {
             throw std::runtime_error("trajectory.getDuration() was infinite");
         }
-        float t = 0.0;
-        constexpr float k_timestep = 0.2F;  // seconds
+        double t = 0.0;
+        constexpr double k_timestep = 0.2;  // seconds
         while (t < duration) {
             Eigen::VectorXd position = trajectory.getPosition(t);
             Eigen::VectorXd velocity = trajectory.getVelocity(t);
             p.push_back(vector6d_t{position[0], position[1], position[2], position[3], position[4], position[5]});
             v.push_back(vector6d_t{velocity[0], velocity[1], velocity[2], velocity[3], velocity[4], velocity[5]});
-            time.push_back(k_timestep);
+            time.push_back((float)k_timestep);
```

Comment:
we convert to float here to not bloat the log files

Reply:
I actually think it is more than that: I think the UR driver actually takes this as float:
- https://github.com/UniversalRobots/Universal_Robots_Client_Library/blob/bff7bf2e2a85c17fa3f88adda241763040596ff1/include/ur_client_library/ur/ur_driver.h#L511